### PR TITLE
Adding NIST algorithm

### PR DESCRIPTION
**Pull request by fidian**
Might be useful for comparing algorithms.

### CODE REVIEW

File: lib/password-strength.js [hunk 1]
```diff
@@ -397,6 +399,72 @@
...
+         * @param {string} password
+         * @return {number}
+         */
+        this.nistScore = function (password) {
+            var len, score;
+
+            len = password.length;
+            score = 0;
+
+            if (len > 20) {
```
Comment (AbsentSemicolon) [inline lib/password-strength.js:436]: Shouldn't we have a test for this for completion's sake at least?

File: lib/password-strength.js [hunk 2]
```diff
@@ -397,6 +399,72 @@
...
+                score += len - 20;
+                len = 20;
+            }
+
+            if (len > 8) {
```
Comment (AbsentSemicolon) [inline lib/password-strength.js:441]: Shouldn't we have a test for this for completion's sake at least?

File: lib/password-strength.js [hunk 3]
```diff
@@ -397,6 +399,72 @@
...
+                score += 2 * (len - 1);
+                len = 1;
+            }
+
+            if (len) {
+                score += 4;
+            }
+
+            /*jslint regexp:true*/
+            if (password.match(/[A-Z]/) && password.match(/[^A-Za-z]/)) {
```
Comment (AbsentSemicolon) [inline lib/password-strength.js:456]: Should we have a test for this to make sure things are working properly? Looks like we only have a few tests for NIST.

**Comment by fidian** [comment]
You're totally right. I've added tests that should cover every condition.

**Comment by AbsentSemicolon** [comment]
Looks good.

**Pull request merged**